examples/mcp-model-context-protocol/main.py

```diff
@@ -0,0 +1,126 @@
+import asyncio
+
+from mcp import ClientSession, StdioServerParameters
+from mcp.client.stdio import stdio_client
+from tensorzero import AsyncTensorZeroGateway, Message, Text, ToolCall, ToolResult
+
+
+async def main():
+    # Define the MCP server
+    server_params = StdioServerParameters(
+        command="uv",
+        args=[
+            "run",
+            "--with",
+            "mcp-clickhouse",
+            "--python",
+            "3.13",
+            "mcp-clickhouse",
+            "2> /dev/null",
+        ],
+        env={
+            "CLICKHOUSE_HOST": "localhost",
+            "CLICKHOUSE_PORT": "8123",
+            "CLICKHOUSE_USER": "chuser",
+            "CLICKHOUSE_DATABASE": "tensorzero",
+            "CLICKHOUSE_PASSWORD": "chpassword",
+            "CLICKHOUSE_SECURE": "false",  # running locally
+        },
+    )
+
+    # Initialize the TensorZero Gateway
+    async with await AsyncTensorZeroGateway.build_http(
+        gateway_url="http://localhost:3000"
+    ) as t0:
+        # Initialize the MCP client
+        async with stdio_client(server_params) as (read, write):
+            async with ClientSession(
+                read,
+                write,
+            ) as session:
+                await session.initialize()
+
+                # Load the tools from the MCP server
+                mcp_tools = await session.list_tools()
+                mcp_tools = [tool.model_dump() for tool in mcp_tools.tools]
+
+                # Convert MCP tool format to TensorZero tool format
```

Comment:
note for future; we might just want to take tools in this format via another body parameter and spare people the couple LOC